packages/utxo-lib/src/transaction/bitcoin.ts

```diff
@@ -74,13 +74,18 @@ export function fromBuffer(buffer: Buffer, options: TransactionOptions) {
 
     const tx = fromConstructor(options);
     tx.version = bufferReader.readInt32();
-
     const marker = bufferReader.readUInt8();
     const flag = bufferReader.readUInt8();
 
-    const hasMweb = flag === MWEB_PEGOUT_TX_FLAG && isNetworkType('litecoin', tx.network);
-
     let hasWitnesses = false;
+    let hasMweb = false;
+
+    if (isNetworkType('litecoin', tx.network)) {
```

Comment:
If I understand it correctly, flag (either mw pegout or segwit) is there only when there is the advanced tx marker (zero byte) as well, but occasionally we interpreted some txin data as mw pegout flag in a pre-segwit transaction as we didn't check the marker first, right?

Reply:
Thats how I understand it too